pkg/proxy/cache.go

```diff
@@ -6,73 +6,62 @@ import (
 
 	"github.com/codeready-toolchain/registration-service/pkg/application"
 	"github.com/codeready-toolchain/registration-service/pkg/log"
-	"github.com/codeready-toolchain/registration-service/pkg/proxy/namespace"
+	"github.com/codeready-toolchain/registration-service/pkg/proxy/access"
 
 	"github.com/gin-gonic/gin"
 )
 
-type UserNamespaces struct {
+type UserAccess struct {
 	sync.RWMutex
 	app application.Application
 
-	// namespace accesses by username
-	namespaces map[string]*namespace.NamespaceAccess
+	// cluster accesses by username
+	clusterAccesses map[string]*access.ClusterAccess
 }
 
-func NewUserNamespaces(app application.Application) *UserNamespaces {
-	return &UserNamespaces{
-		app:        app,
-		namespaces: make(map[string]*namespace.NamespaceAccess),
+func NewUserAccess(app application.Application) *UserAccess {
+	return &UserAccess{
+		app:             app,
+		clusterAccesses: make(map[string]*access.ClusterAccess),
 	}
 }
 
-// Get tries to retrieve the namespace access from the cache first. If found then it checks if the cached access is still valid.
-// If not found or invalid then retrieves a new namespace access from the member cluster service and stores it in the cache.
-func (c *UserNamespaces) Get(ctx *gin.Context, userID, username string) (*namespace.NamespaceAccess, error) {
-	na, err := c.namespaceFromCache(ctx, username)
-	if err != nil {
-		return nil, err
-	}
-	if na != nil {
-		return na, nil
+// Get tries to retrieve the cluster access from the cache first. If found then it checks if the cached access is still valid.
+// If not found or invalid then retrieves a new cluster access from the member cluster service and stores it in the cache.
+func (c *UserAccess) Get(ctx *gin.Context, userID, username string) (*access.ClusterAccess, error) {
+	ca := c.clusterAccessFromCache(ctx, username)
+	if ca != nil {
+		return ca, nil
 	}
 
-	// Not found in cache or invalid. Get a new namespace access.
-	log.Info(ctx, fmt.Sprintf("Retrieving a new Namespace Access from the cluster for the user : %s", username))
-	na, err = c.app.MemberClusterService().GetNamespace(ctx, userID, username)
+	// Not found in cache or invalid. Get a new cluster access.
+	log.Info(ctx, fmt.Sprintf("Retrieving a new Cluster Access from the cluster for the user : %s", username))
+	ca, err := c.app.MemberClusterService().GetClusterAccess(ctx, userID, username)
 	if err != nil {
 		return nil, err
 	}
 	// Store in the cache
-	c.add(username, na)
+	c.add(username, ca)
 
-	return na, nil
+	return ca, nil
 }
 
-// namespaceFromCache tries to retrieve a namespace access from the cache and validates it.
-// Returns nil, nil if no namespace access found or if it's invalid.
-func (c *UserNamespaces) namespaceFromCache(ctx *gin.Context, username string) (*namespace.NamespaceAccess, error) {
+// clusterAccessFromCache tries to retrieve a cluster access from the cache and validates it.
```

Comment:
Same here. No validation anymore.

Reply:
Good catch, done in d36401dd79ba7c2e9467cd07278381699783fd21